core/ArchiveProcessor/Loader.php

```diff
@@ -508,6 +508,19 @@ public function canSkipThisArchive()
             && !$hasChildArchivesInPeriod;
     }
 
+    private function hasChildArchivesInPeriod($idSite, Period $period): bool
+    {
+        $cacheKey = CacheId::siteAware('Archiving.hasChildArchivesInPeriod.' . $period->getRangeString());
```

Comment:
```suggestion
        $cacheKey = CacheId::siteAware('Archiving.hasChildArchivesInPeriod.' . $period->getRangeString(), [$idSite]);
```

All (three) site aware cache keys need to receive an explicit site id.

The automatic detection does not always work, as both `CronArchive` and the `QueueConsumer` don't have the required parameters set. And then the cache key will be created without any site information.

Running CLI archiving for multiple sites will reuse the cached values of the first site for all other sites. If the first site has no visits for a day that is something we really do not want.

Reply:
Ha, interesting. I was thinking about it initially but then expected the auto-detection would be sufficient. Added to all 3 calls.